### PR TITLE
Add erlang compatibility matrix

### PR DESCRIPTION
**Pull request by conradwt**
This PR supports the following features:

- add Erlang compatibility matrix

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,53 +53,60 @@ The installation process for ElixirLS depends on your editor.
...
+- 1.12 or newer
 
 Erlang:
 
-- OTP 22 minimum
+| Versions | Supported |                          Issue(s)                          |
+| :------: | :-------: | :--------------------------------------------------------: |
+|  OTP 22  |    Yes    |                            None                            |
+|  OTP 23  |    Yes    |                            None                            |
+|  OTP 24  |    Yes    |                            None                            |
```
Comment (lukaszsamson) [inline README.md:107]:
```suggestion
|  OTP 24  |    Yes    |         Elixir 1.12 is not supported         |
```

File: README.md [hunk 2]
```diff
@@ -53,53 +53,60 @@ The installation process for ElixirLS depends on your editor.
...
 
-- 1.12 minimum
+- 1.12 or newer
 
 Erlang:
 
-- OTP 22 minimum
+| Versions | Supported |                          Issue(s)                          |
+| :------: | :-------: | :--------------------------------------------------------: |
+|  OTP 22  |    Yes    |                            None                            |
```
Comment (lukaszsamson) [inline README.md:105]:
```suggestion
|  OTP 22  |    Yes    |                            Erlang docs not working (requires EIP 48)                            |
```

File: README.md [hunk 3]
```diff
@@ -53,53 +53,54 @@ The installation process for ElixirLS depends on your editor.
...
-- 1.12 minimum
-
-Erlang:
-
-- OTP 22 minimum
+| OTP Versions | Elixir Versions | Supports ElixirLS |                          Issue(s)                          |
+| :----------: | :-------------: | :---------------: | :--------------------------------------------------------: |
+|      22      |   1.12 - 1.13   |        Yes        |         Erlang docs not working (requires EIP 48)          |
+|      23      |   1.12 - 1.14   |        Yes        |                            None                            |
+|      24      |   1.12 - 1.15   |        Yes        |                            None                            |
```
Comment (lukaszsamson) [inline README.md:101]:
```suggestion
|      24      |   1.13 - 1.14   |        Yes        |                            None                            |
```

Comment (conradwt) [reply README.md:101]: OTP 26 is only supported by Elixir 1.14.5 - 1.15.0 at this time.  In your updates, you're specifying `any` Elixir version for OTP 26.0.  BTW, I referenced the following Elixir docs:

- https://hexdocs.pm/elixir/1.14.5/compatibility-and-deprecations.html#compatibility-between-elixir-and-erlang-otp

- https://hexdocs.pm/elixir/1.15.0-rc.1/compatibility-and-deprecations.html#compatibility-between-elixir-and-erlang-otp

Comment (lukaszsamson) [reply README.md:101]: I wanted to express the fact that 1.15 is not supported yet. Do you think it can be improved?

Comment (conradwt) [reply README.md:101]: I understand.  Do you think that we can simply use `none` instead of `any` in regard to the Elixir version?

**Pull request merged**